chill-thrift/src/main/java/com/twitter/chill/thrift/TBaseSerializer.java

```diff
@@ -53,9 +53,9 @@ public void write(Kryo kryo, Output output, TBase tBase) {
   }
 
   @Override
-  public TBase read(Kryo kryo, Input input, Class<TBase> tBaseClass) {
+  public TBase read(Kryo kryo, Input input, Class<? extends TBase> tBaseClass) {
     try {
-      TBase prototype = tBaseClass.newInstance();
+      TBase prototype = tBaseClass.getDeclaredConstructor().newInstance();
```

Comment:
why this change out of curiosity?

Reply:
`newInstance()` was marked deprecated by my IDE. This seemed to the the recommended replacement. I think it has to do with `newInstance()` bypassing a checked exception.

https://www.xspdf.com/resolution/53014482.html
https://stackoverflow.com/a/53014482/118999

There were some other deprecated APIs and solutions, but I wasn't sure what versions of Scala Chill is used with so I didn't make other changes. I did see some versions in the `build.sbt`, so would a proper test encompass building with each of those Scala versions?

Reply:
Also, when I ran `publishM2` to generate local artifacts, I didn't get a `chill-java` to use in my Spark testing. But when I changed the Scala version it did build. Could you give me a quick guideline for how I should be testing this library? Do I need to modify the `build.sbt` to check each Scala version?

Reply:
Okay, I'd rather separate changes that are core to the kryo5 change and those irrelevant.

As far as testing, there are the tests in the repo (sbt test), but if you want to do integration tests with spark, you will need to do the publish. I'm actually not a huge sbt expert to share much there.

You can set the scala version in sbt by doing: `sbt ++2.12.12 publishM2`, for instance.

Reply:
I reverted this change as requested.